src/components/CompareResults/TableHeader.tsx

```diff
@@ -32,7 +33,20 @@ function FilterableColumn({
   onToggle,
   onClear,
 }: FilterableColumnProps) {
+  const filterCountStyle = style({
+    paddingLeft: `${Spacing.xSmall}px`,
+  });
   const popupState = usePopupState({ variant: 'popover', popupId: columnId });
+  //this returns number of possible values that are checked
+  const possibleCheckedValues = possibleValues.reduce(
+    (count: number, value) => {
+      if (!uncheckedValues?.has(value)) {
+        count++;
+      }
+      return count;
+    },
+    0,
+  );
```

Comment:
dosn't this work? `possibleValues.length - uncheckedValues.size`

Reply:
Yes, that makes sense but `uncheckedValues.size` is possible undefined so I went the longest route. My brain likes the longest route too much lol. 

However this works:
```
  const possibleCheckedValues =
    possibleValues.length - (uncheckedValues?.size || 0);
```

Reply:
Also I kind of would like to reverse the meaning of what's in the set now... I think it was a bad idea of mine to have in the set what's been deselected as opposed to what's been selected (I think it was easier to implement this way at the time).

We can do it afterwards.

Reply:
Yes, I was initially confused by the unchecked values vs checked but how it is now makes sense because by default all the values are checked vs an opposite user flow where by default the values are unchecked.

Reply:
@julienw But now I'm questioning why they're all checked by default. It's like all the filters _are on_ which is an uncommon user flow, no? Usually, by default, the filter is unchecked, then you check it, and it filters the table. 

Also, why do we have two options in the dropdown that do the same thing? This needs revisiting. 

Example: "Select only Windows"  and the filter option where I've only selected Windows. 

<img width="208" alt="Screenshot 2024-11-22 at 11 24 44" src="https://github.com/user-attachments/assets/e03575bf-393c-4650-91a7-7aa9cd9e6fec">

Reply:
Also, we might need to add checkboxes here too like in the original figma design. That might make it less confusing. I'll file a ticket.

![image](https://github.com/user-attachments/assets/938f6cb0-9fc0-44cc-b7b4-22486a44cb5d)

Reply:
> Also, why do we have two options in the dropdown that do the same thing? This needs revisiting.
> Example: "Select only Windows" and the filter option where I've only selected Windows.

Not the same thing :-)

"Select only Windows" will deselect everything else where needed.
But clicking just "Windows" will keep the other options.